raft/raft.go

```diff
@@ -663,6 +663,11 @@ func stepFollower(r *raft, m pb.Message) {
 }
 
 func (r *raft) handleAppendEntries(m pb.Message) {
+	if m.Index < r.Commit {
```

Comment:
Could this be done in log.maybeAppend instead of a special case here? Just replace the panic with `return l.committed, true`